amd_openvx_extensions/amd_nn/src/slice_layer.cpp

```diff
@@ -1,345 +1,393 @@
+    if(status) {
+        std::cerr << "ERROR: vxMapTensorPatch() failed for tmpi tensor (" << status << ")" << std::endl;
+        return -1;
+    }
+    for(int i=0; i<icount; i++) {
+        fptr[i] = input[i];
+    }
+    status = vxUnmapTensorPatch(tmpi_tensor, map_id);
 
-//! \brief The OpenCL code generator callback.
-static vx_status VX_CALLBACK opencl_codegen(
-        vx_node node,                                  // [input] node
-        const vx_reference parameters[],               // [input] parameters
-        vx_uint32 num,                                 // [input] number of parameters
-        bool opencl_load_function,                     // [input]  false: normal OpenCL kernel; true: reserved
-        char opencl_kernel_function_name[64],          // [output] kernel_name for clCreateKernel()
-	std::string& opencl_kernel_code,               // [output] string for clCreateProgramWithSource()
-	std::string& opencl_build_options,             // [output] options for clBuildProgram()
-	vx_uint32& opencl_work_dim,                    // [output] work_dim for clEnqueueNDRangeKernel()
-	vx_size opencl_global_work[],                  // [output] global_work[] for clEnqueueNDRangeKernel()
-	vx_size opencl_local_work[],                   // [output] local_work[] for clEnqueueNDRangeKernel()
-	vx_uint32& opencl_local_buffer_usage_mask,     // [output] reserved: must be ZERO
-	vx_uint32& opencl_local_buffer_size_in_bytes   // [output] reserved: must be ZERO
-    )
-{
-    //get tensor dimensions
-    vx_size input_dims[4];
-    vx_size op_size_per_batch[8], batch_size = 0, ip_batch_stride = 0;
-    vx_enum type;
+    input_tensor[0]  = tmpi_tensor;
 
-    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DIMS, input_dims, sizeof(input_dims)));
-    ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[0], VX_TENSOR_DATA_TYPE, &type, sizeof(type)));
-    batch_size = input_dims[3];
-    ip_batch_stride = input_dims[2] * input_dims[1] * input_dims[0];
-#if ENABLE_DEBUG_PRINT_DIMS
-    std::cout << "slice input " << input_dims[3] << " " << input_dims[2] << " " << input_dims[1] << " " << input_dims[0] << std::endl;
-#endif
-
-    int i = 1;
-    while(parameters[i] && (i < 9)) {
-        vx_size output_dims[4];
-        ERROR_CHECK_STATUS(vxQueryTensor((vx_tensor)parameters[i], VX_TENSOR_DIMS, output_dims, sizeof(output_dims)));
-        op_size_per_batch[i-1] = output_dims[2] * output_dims[1] * output_dims[0];
-#if ENABLE_DEBUG_PRINT_DIMS
-        std::cout << "slice output " << i << " " << output_dims[3] << " " << output_dims[2] << " " << output_dims[1] << " " << output_dims[0] << std::endl;
-#endif
-
-        i++;
+    // copy starts index
+    status = vxMapTensorPatch((vx_tensor)parameters[2], 1, nullptr, nullptr, &map_id, stride, (void **)&ptr, VX_READ_ONLY, VX_MEMORY_TYPE_HOST, 0);
```

Comment:
Codacy found an issue: [Variable 'status' is reassigned a value before the old one has been used.](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=5932170)